node/src/effect/console.rs

```diff
@@ -0,0 +1,60 @@
+use std::{
+    borrow::Cow,
+    fmt::{Debug, Display},
+};
+
+use casper_types::EraId;
+use datasize::DataSize;
+use futures::Future;
+use serde::Serialize;
+
+use super::Responder;
+use crate::components::consensus::EraDump;
+
+/// A request to dump the internal consensus state of a specific era.
+#[derive(DataSize, Serialize)]
+pub(crate) struct DumpConsensusStateRequest {
+    /// Era to serialize.
+    ///
+    /// If not given, use active era.
+    pub(crate) era_id: Option<EraId>,
+    /// Serialization function to serialize the given era with.
+    #[data_size(skip)]
+    #[serde(skip)]
+    pub(crate) serialize: fn(&EraDump) -> Result<Vec<u8>, Cow<'static, str>>,
```

Comment:
Can you explain to me why it has to be `Cow`?

Reply:
I think `Cow<'static, str>` is pretty standard (or should be). If need be, a neat trick would be to `type Str = Cow<'static, str>' to make it more readable.

The main reason is to not have to write `"asdf".to_string()` further down.

Reply:
Why not `Result<Vec<u8>, String>`?

Reply:
That's what I'm trying to avoid. It makes it so that you have to heap allocate a copy of every fixed string returned as an error (`return Err("missing foo".to_owned()))`).

Reply:
That's not really a big performance issue, isn't it? Short string only in case of errors which should be infrequent anyway.